src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java

```diff
@@ -179,6 +183,7 @@ private void renew(RoutingContext routingContext) {
         automatedPatronBlocksValidator, r, errorHandler, USER_IS_BLOCKED_AUTOMATICALLY))
       .thenComposeAsync(r -> refuseIfNoPermissionsForRenewalOverride(
         overrideRenewValidator, r, errorHandler))
+      .thenCompose(r -> r.after(ctx -> lookupOverdueFinePolicy(ctx, overdueFinePolicyRepository, errorHandler)))
```

Comment:
That's potentially a second overdue policy lookup during renew, a second API call to the same endpoint with the same result. Check OverdueFineService. Considering that the context now has this policy, we should probably just use it in OverdueFineService instead of making a second call.

Reply:
Makes sense but is complicated by the fact that CheckIn also uses OverdueFineService and doesn't put the OverdueFinePolicy on the loan in CheckInContext. 

The smallest, but probably not the prettiest, change is to have OverdueFineService branch out on the type of context: 

` private CompletableFuture<Result<FeeFineAction>> createOverdueFineIfNecessary(Loan loan,
    Scenario scenario, String loggedInUserId) {

    log.debug("createOverdueFineIfNecessary:: parameters loan: {}, scenario: {}, " +
      "loggedInUserId: {}", () -> loan, () -> scenario, () -> loggedInUserId);

    if (scenario.equals(Scenario.CHECKIN)) {
      return overdueFinePolicyRepository.findOverdueFinePolicyForLoan(succeeded(loan))
        .thenCompose(r -> r.after(l -> scenario.shouldCreateFine(l.getOverdueFinePolicy())
          ? createOverdueFine(l, loggedInUserId)
          : completedFuture(succeeded(null))));
    } else {
      return scenario.shouldCreateFine(loan.getOverdueFinePolicy())
        ? createOverdueFine(loan, loggedInUserId)
        : completedFuture(succeeded(null));
    }
  }`

Would you like to have CheckIn process put the OverdueFinePolicy on the context too, or apply something like the change above to OverdueFineService, or just leave it be, or something fourth?

Reply:
(well, except that, that sample code wouldn't actually work as written)

Reply:
I think that's ok, it also makes sense to extract these lines to a method to avoid code duplication:
```
scenario.shouldCreateFine(overdueFinePolicy)
    ? createOverdueFine(loan, loggedInUserId)
    : completedFuture(succeeded(null));
```

Reply:
Okay, done.